src/server/session.ts

```diff
@@ -677,6 +679,45 @@ module ts.server {
             }
         }
 
+        getDiagnosticsForProject(delay: number, fileName: string) {
+            let { configFileName, fileNameList } = this.getProjectInfo(fileName, true);
+            fileNameList = fileNameList.filter((value, index, array) => value.indexOf("lib.d.ts") < 0);
+
+            // Sort the file name list to make the recently touched files come first
+            let highPriorityFiles: string[] = [];
+            let mediumPriorityFiles: string[] = [];
+            let lowPriorityFiles: string[] = [];
+            let veryLowPriorityFiles: string[] = [];
+            let normalizedFilePath = ts.normalizePath(fileName);
+            let project = this.projectService.getProjectForFile(normalizedFilePath);
+            for (let oneFile of fileNameList) {
+                if (oneFile == normalizedFilePath)
+                    highPriorityFiles.push(oneFile);
+                else {
+                    let info = this.projectService.getScriptInfo(oneFile);
+                    if (!info || !info.isOpen) {
+                        if (oneFile.indexOf(".d.ts") > 0)
+                            veryLowPriorityFiles.push(oneFile);
+                        else
+                            lowPriorityFiles.push(oneFile);
+                        this.projectService.openFile(oneFile, true);

```

Comment:
i do not think we need this.. the editor will send open messages when the file is open.